test/Microsoft.AspNetCore.Razor.Test.Common/Language/Legacy/IntializeTestFileAttribute.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.Language.Legacy
+{
+    public class IntializeTestFileAttribute : BeforeAfterTestAttribute
+    {
+        private static int _callCount = 0;
+        private static string _previousMethod;
```

Comment:
You will need to use `AsyncLocal` for these. That's likely why you're having threading issues.